includes/api/ApiQueryRevisions.php

```diff
@@ -348,6 +378,19 @@ public function execute() {
 		$count = 0;
 		$res = $this->select( __METHOD__ );
 
+		// SUS-3118: Issue a single bulk query for user name info if needed
+		if ( $this->fld_user ) {
+			$userIds = [];
+			foreach ( $res as $row ) {
+				if ( $row->rev_user ) {
+					$userIds[] = $row->rev_user;
+				}
+			}
+			$res->rewind();
+
+			$this->userNames = User::whoAre( $userIds );
+		}
+
 		foreach ( $res as $row ) {
 			if ( ++ $count > $limit ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this increment or decrement operator into a dedicated statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS881)

[project-id]: wikia:app